clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1231,19 +1244,32 @@ public void seekToEnd(Collection<TopicPartition> partitions) {
     public long position(TopicPartition partition) {
         acquire();
         try {
-            if (!this.subscriptions.isAssigned(partition))
-                throw new IllegalArgumentException("You can only check the position for partitions assigned to this consumer.");
-            Long offset = this.subscriptions.position(partition);
-            if (offset == null) {
-                updateFetchPositions(Collections.singleton(partition));
-                offset = this.subscriptions.position(partition);
-            }
-            return offset;
+            positions(Collections.singleton(partition));
+            return this.subscriptions.position(partition);
         } finally {
             release();
         }
     }
 
+    private void positions(Set<TopicPartition> partitions) {
```

Comment:
It seems a little weird to have no return value for `positions()` while `position()` returns a long. Would it be better to name it something like `updatePositions()`?